sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -165,19 +177,32 @@ class FileStreamSourceSuite extends FileStreamSourceTest with SharedSQLContext {
       .collect { case s @ StreamingRelation(dataSource, _, _) => s.schema }.head
   }
 
-  test("FileStreamSource schema: no path") {
+  // ============= Basic parameter exists tests ================
+
+  test("FileStreamSource schema: no path, no schema") {
     val e = intercept[IllegalArgumentException] {
       createFileStreamSourceAndGetSchema(format = None, path = None, schema = None)
     }
-    assert("'path' is not specified" === e.getMessage)
+    assert(e.getMessage.contains("path"))  // reason is path, not schema
   }
 
-  test("FileStreamSource schema: path doesn't exist") {
-    intercept[AnalysisException] {
+  test("FileStreamSource schema: path doesn't exist, no schema") {
+    val e = intercept[IllegalArgumentException] {
       createFileStreamSourceAndGetSchema(format = None, path = Some("/a/b/c"), schema = None)
     }
```

Comment:
@marmbrus 
This test failed earlier with "path does not exist" because the system tried to infer the schema and failed as path did not exists. If the schema is specified, then there is not failure if path does not exist. 

I feel that the behavior of file streams when the path does not exist, should not depend on whether schema inference is enabled or not. So there is the question "Should file stream creation fail when path does not exist, or not?"